hummingbot/client/hummingbot_application.py

```diff
@@ -1139,6 +1140,85 @@ def compare_balance_snapshots(self):
         lines = ["", "  Performance:"] + ["    " + line for line in str(df).split("\n")]
         self._notify("\n".join(lines))
 
+    def add_balances(self, msp, asset_type, asset_time, base, quote):
+        market_name = msp.market.name
+
+        if asset_type == "base":
+            asset_list = base
+            asset = msp.base_asset
+        elif asset_type == "quote":
+            asset_list = quote
+            asset = msp.quote_asset
+        else:
+            raise ValueError("asset_type must be either 'base' or 'quote'")
+
+        if asset_time == "starting":
+            amount = self.starting_balances.get(asset).get(market_name)
+        elif asset_time == "current":
+            amount = self.balance_snapshot().get(asset).get(market_name)
+        else:
+            raise ValueError("asset_time must be either 'starting' or 'current'")
+
+        if len(asset_list) == 0:
+            asset_list.insert(0, asset)
+            asset_list.insert(1, float(amount))
+        else:
+            if asset_list[0] == asset:
+                asset_list[1] += float(amount)
+            else:
+                erc = ExchangeRateConversion.get_instance()
+                asset_list[1] += float(erc.convert_token_value(
+                    amount=float(amount),
+                    from_currency=asset,
+                    to_currency=asset_list[0]
+                ))
+        if asset_type == "base":
+            base = asset_list
+        elif asset_type == "quote":
+            quote = asset_list
+
+        return base, quote
+
+    def analyze_performance(self):
+        if len(self.starting_balances) == 0:
+            self._notify("  Performance analysis is not available before bot starts")
+            return
+
+        starting_base = list()
+        starting_quote = list()
+        current_base = list()
+        current_quote = list()
+
+        for msp in self.market_symbol_pairs:
+            for asset_type in ["base", "quote"]:
+                starting_base, starting_quote = self.add_balances(
+                    msp=msp,
+                    asset_type=asset_type,
+                    asset_time="starting",
+                    base=starting_base,
+                    quote=starting_quote
+                )
+                current_base, current_quote = self.add_balances(
+                    msp=msp,
+                    asset_type=asset_type,
+                    asset_time="current",
+                    base=current_base,
+                    quote=current_quote
+                )
+
+        market_pair_info = self.market_symbol_pairs[0]
```

Comment:
why are we only doing this for the first pair?

Reply:
We use the first market_symbol_pair because if the trading pairs are different, such as WETH-DAI and ETH-USD, the tuples above will contain the information in terms of the first trading pair based on the way that the balances were computed in the for loop above.

Reply:
But wouldn't you miss the changes in the ETH-USD pair?

Reply:
In this case, I would convert the values of ETH and USD into WETH and DAI to have one "combined" balance across both exchanges.